accounts/tests.py

```diff
@@ -378,32 +393,165 @@ def test_failure_post_with_incorrect_user(self):
 
 
 class TestFollowView(TestCase):
+    def setUp(self):
+        user = {
+            "email": "test@example.com",
+            "username": "test",
+            "password1": "goodpass",
+            "password2": "goodpass",
+        }
+        user2 = {
+            "email": "test@example.com",
+            "username": "second",
+            "password1": "goodpass",
+            "password2": "goodpass",
+        }
+        self.client.post(reverse("accounts:signup"), user)
+        User.objects.create_user(user2["username"], user2["email"], user2["password1"])
+        self.user = User.objects.get(username="test")
+        self.user2 = User.objects.get(username="second")
+
     def test_success_post(self):
-        pass
+        response = self.client.post(
+            reverse("accounts:follow", kwargs={"username": "second"}), None
+        )
+        self.assertRedirects(
+            response,
+            reverse("accounts:home"),
+            status_code=302,
+            target_status_code=200,
+            msg_prefix="",
+            fetch_redirect_response=True,
+        )
+        self.assertTrue(FriendShip.objects.filter(follower=self.user2).exists())
 
     def test_failure_post_with_not_exist_user(self):
-        pass
+        response = self.client.post(
+            reverse("accounts:follow", kwargs={"username": "third"}), None
+        )
+        self.assertEqual(response.status_code, 404)
+        self.assertFalse(FriendShip.objects.filter(follower=self.user2).exists())
 
     def test_failure_post_with_self(self):
-        pass
+        response = self.client.post(
+            reverse("accounts:follow", kwargs={"username": "test"}), None
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertFalse(FriendShip.objects.filter(followee=self.user).exists())
+
+    def test_failure_post_with_already_followed_user(self):
+        FriendShip.objects.create(followee=self.user, follower=self.user2)
+        response = self.client.post(
+            reverse("accounts:follow", kwargs={"username": "test"}), None
+        )
+        self.assertEqual(response.status_code, 200)
 
 
 class TestUnfollowView(TestCase):
+    def setUp(self):
+        user = {
+            "email": "test@example.com",
+            "username": "test",
+            "password1": "goodpass",
+            "password2": "goodpass",
+        }
+        user2 = {
+            "email": "test@example.com",
+            "username": "second",
+            "password1": "goodpass",
+            "password2": "goodpass",
+        }
+        self.client.post(reverse("accounts:signup"), user)
+        User.objects.create_user(user2["username"], user2["email"], user2["password1"])
+        self.user = User.objects.get(username="test")
+        self.user2 = User.objects.get(username="second")

```

Comment:
```suggestion
self.user = User.objects.create_user(username="test", email="test@test.com", password="goodpass")
self.user2 = User.objects.create_user(username="test2", email="test2@test.com", password="goodpass")
self.client.login(username="test", password="goodpass")
```
createの段階でインスタンス化できますので、他のところもこのようにして貰えばと思います

Reply:
修正しました。

Reply:
修正しました。